content/blog/deploy-stateful-mongodb-application-on-kubernetes-clusters-in-hpe-greenlake-for-containers-1.md

```diff
@@ -0,0 +1,547 @@
+---
+title: Deploy Stateful MongoDB Application on Kubernetes Clusters in HPE
+  GreenLake for Containers
+date: 2022-08-16T14:02:17.749Z
+author: Guoping Jia
+authorimage: /img/guoping.png
+tags:
+  - hpe-greenlake
+  - kubernetes
+  - mongodb
+---
+## Introduction
+
+[HPE GreenLake for Containers](https://www.hpe.com/us/en/greenlake/containers.html), one of the HPE GreenLake cloud services available on the HPE GreenLake Central platform, allows customers to open the Clusters screen to create a cluster, view details about existing clusters, and launch the HPE GreenLake for Container service console. It provides an enterprise-grade container management service using open source Kubernetes. 
+
+This blog post describes the process of deploying the stateful MongoDB application on the created Kubernetes clusters in HPE GreenLake for Containers. Using the Kubernetes StatefulSet and Headless Service, together with the pre-provisioned persistent volumes, the MongoDB application can be deployed as a Replica Set that provides redundancy, fault tolerance and high availability. This MongoDB application deployment can be used for working with a large amount of data and high loads in customer production environments.
+
+## MongoDB Application
+
+[MongoDB](https://www.mongodb.com/) is a general-purpose, document-based NoSQL database program which is a popular alternative to traditional databases. The MongoDB data model can represent any kind of object structures which can have properties or even be nested for multiple levels. Unlike the relational databases, MongoDB can store large amounts of data without requiring a logical category or schema. Therefore, it takes up the challenge of focusing on scalability and the speed of queries. MongoDB is free, open-source software. You can download the MongoDB packages, set them up and configure them at no expense.
```

Comment:
Change "NoSQL database program which is..." to "NoSQL database program that is..."
Similarly, in the next sentence, change "...object structures which can..." to "...object structures that can"
(There's often a lot of confusion as to when to use "that" and "which". This link might be helpful for future blog posts:
https://www.diffen.com/difference/That_vs_Which#:~:text=%22That%22%20is%20used%20to%20indicate,the%20end%20of%20the%20clause.

Reply:
Thank you for pointing me this helpful link. Good learning :-)